incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -33,13 +36,16 @@ const (
 type Hierarchy struct {
 	Log     logr.Logger
 	Forest  *forest.Forest
-	authz   authzClient
+	server  serverClient
 	decoder *admission.Decoder
 }
 
-// authzClient represents the authz checks that should typically be performed against the apiserver,
-// but need to be stubbed out during unit testing.
-type authzClient interface {
+// serverClient represents the checks that should typically be performed against the apiserver, but
+// need to be stubbed out during unit testing.
+type serverClient interface {
+	// Exists returns true if the given namespace exists.
+	Exists(ctx context.Context, nnm string) (bool, error)
```

Comment:
One nit after reviewing the code: it may be clearer to rename this function to differentiate from `forest.ns.Exists`. I am actually not sure after another scan. Maybe it's fine to leave it as is.